src/core/components/hv-root/index.tsx

```diff
@@ -28,6 +48,460 @@ export default class Hyperview extends PureComponent<HvScreenProps.Props> {
+
+  constructor(props: HvScreenProps.Props) {
+    super(props);
+
+    this.onUpdate = this.onUpdate.bind(this);
+    this.reload = this.reload.bind(this);
+
+    this.parser = new Dom.Parser(
+      this.props.fetch,
+      this.props.onParseBefore,
+      this.props.onParseAfter,
+    );
+
+    this.behaviorRegistry = Behaviors.getRegistry(this.props.behaviors);
+    this.componentRegistry = Components.getRegistry(this.props.components);
+    this.formComponentRegistry = Components.getFormRegistry(
+      this.props.components,
+    );
+  }
+
+  /**
+   * Reload if an error occured.
+   * @param opt_href: Optional string href to use when reloading the screen. If not provided,
+   * the screen's current URL will be used.
+   */
+  reload = (
+    optHref: DOMString | null | undefined,
+    opts: HvComponentOptions,
+    callbacks: OnUpdateCallbacks,
+  ) => {
+    const isBlankHref =
+      optHref === null ||
+      optHref === undefined ||
+      optHref === '#' ||
+      optHref === '';
+    const url = isBlankHref
+      ? callbacks.getStateUrl()
+      : UrlService.getUrlFromHref(optHref, callbacks.getStateUrl());
+
+    if (!url) {
+      return;
+    }
+
+    const options = opts || {};
+    const {
+      behaviorElement,
+      showIndicatorIds,
+      hideIndicatorIds,
+      once,
+      onEnd,
+    } = options;
+
+    const showIndicatorIdList = showIndicatorIds
+      ? Xml.splitAttributeList(showIndicatorIds)
+      : [];
+    const hideIndicatorIdList = hideIndicatorIds
+      ? Xml.splitAttributeList(hideIndicatorIds)
+      : [];
+
+    if (once && behaviorElement) {
+      if (behaviorElement.getAttribute('ran-once')) {
+        // This action is only supposed to run once, and it already ran,
+        // so there's nothing more to do.
+        if (typeof onEnd === 'function') {
+          onEnd();
+        }
+        return;
+      }
+      Behaviors.setRanOnce(behaviorElement);
+    }
+
+    let newRoot = callbacks.getDoc();
+    if (showIndicatorIdList || hideIndicatorIdList) {
+      newRoot = Behaviors.setIndicatorsBeforeLoad(
+        showIndicatorIdList,
+        hideIndicatorIdList,
+        newRoot,
+      );
+    }
+
+    // Re-render the modifications
+    callbacks.setNeedsLoad();
+    callbacks.setState({
+      doc: newRoot,
+      elementError: null,
+      error: null,
+      url,
+    });
+  };
+
+  /**
+   * Fetches the provided reference.
+   * - If the references is an id reference (starting with #),
+   *   returns a clone of that element.
+   * - If the reference is a full URL, fetches the URL.
+   * - If the reference is a path, fetches the path from the host of the URL
+   *   used to render the screen.
+   * Returns a promise that resolves to a DOM element.
+   */
+  fetchElement = async (
+    href: DOMString | null | undefined,
+    method: DOMString | null | undefined = Dom.HTTP_METHODS.GET,
+    root: Document,
+    formData: FormData | null | undefined,
+    callbacks: OnUpdateCallbacks,
+  ) => {
```

Comment:
let's type the returned value:
```suggestion
  ): Promise<Element | null> => {
```